src/gc.c

```diff
@@ -802,11 +812,17 @@ void jl_gc_force_mark_old(jl_ptls_t ptls, jl_value_t *v) JL_NOTSAFEPOINT
         jl_gc_queue_root(v);
 }
 
-#define should_collect() (__unlikely(gc_num.allocd>0))
-
 static inline int maybe_collect(jl_ptls_t ptls)
 {
-    if (should_collect() || gc_debug_check_other()) {
+    int should_collect = 0;
+    if (ptls->gc_num.allocd >= 0) {
```

Comment:
This function need to use atomic relaxed load.

Reply:
Why?

Reply:
Hmm, the line number shows up really weird.... The comment was for `combine_thread_gc_counts`.

(Actually, to avoid UB, all access of these outside of the collection phase should be atomic.)